src/parser/Statement.ts

```diff
@@ -1571,11 +1568,11 @@ export class ClassStatement extends Statement implements TypedefProvider, Symbol
         return constructorFuncType;
     }
 
-    public buildSymbolTable(parentClass?: ClassStatement) {
+    public buildSymbolTable(parentClass?: Statement) {
         this.symbolTable.clear();
-        this.symbolTable.addSymbol('m', this.name?.range, this.getCustomType());
+        this.symbolTable.addSymbol('m', this.name?.range, this.getThisBscType());
         this.memberTable.clear();
-        if (parentClass) {
+        if (isClassStatement(parentClass)) {
```

Comment:
This is there because `buildSymbolTable` is part of an `interface`, and so the typing will work inside this if statement (` parentClass.getConstructorFunctionType()`